rewrite-core/src/test/java/org/openrewrite/text/FindTest.java

```diff
@@ -111,4 +111,106 @@ void caseInsensitive() {
           )
         );
     }
+
+    @Test
+    void regexBasicMultiLine() {
+        rewriteRun(
+          spec -> spec.recipe(new Find("[T\\s]", true, true, true, null, null)),
+          text(
+            """
+              This is\ttext.
+              This is\ttext.
+              """,
+            """
+              ~~>This~~> is~~>\ttext.~~>
+              ~~>This~~> is~~>\ttext.
+              """
+          )
+        );
+    }
+
+    @Test
+    void regexWithoutMultilineAndDotall() {
+        rewriteRun(
+          spec -> spec.recipe(new Find("^This.*below\\.$", true, true, false, false, null)),
+          text(
+            """
+              This is text.
+              This is a line below.
+              This is a line above.
+              This is text.
+              This is a line below.
+              """
+          )
+        );
+    }
+
+    @Test
+    void regexWithoutMultilineAndWithDotAll() {
+        rewriteRun(
+          spec -> spec.recipe(new Find("^This.*below\\.$", true, true, false, true, null)),
+          text(
+            """
+              This is text.
+              This is a line below.
+              This is a line above.
+              This is text.
+              This is a line below.
+              """,
+            """
+              ~~>This is text.
+              This is a line below.
+              This is a line above.
+              This is text.
+              This is a line below.
+              """
+          )
+        );
+    }
+
+    @Test
+    void regexWithMultilineAndWithoutDotall() {
+        rewriteRun(
+          spec -> spec.recipe(new Find("^This.*below\\.$", true, true, true, false, null)),
+          text(
+            """
+              This is text.
+              This is a line below.
+              This is a line above.
+              This is text.
+              This is a line below.
+              """,
+            """
+              This is text.
+              ~~>This is a line below.
+              This is a line above.
+              This is text.
+              ~~>This is a line below.
+              """
+          )
+        );
+    }
+
+    @Test
+    void regexWithBothMultilineAndDotAll() {
+        rewriteRun(
+          spec -> spec.recipe(new Find("^This.*below\\.$", true, true, true, true, null)),
+          text(
+            """
+              This is text.
+              This is a line below.
+              This is a line above.
+              This is text.
+              This is a line below.
+              """,
+            """
+              ~~>This is text.
+              This is a line below.
+              This is a line above.
+              This is text.
+              This is a line below.
+              """
+          )
+        );
+    }
```

Comment:
I find it quite hard to gauge from regex and output what we intent to match here and why; is it the first below? the second below? The first This? any of the others?

Can we use a clearer example with unique values across the lines, maybe something like
```
Zero
One
Two
Three
```
And then match on either `^One.Two$`, or `^One\\sTwo$` with variations of dot all and multilne enabled/disabled. I think that'll be clearer what exactly is matched as we only show the marker start.